cmake/GenerateScripts.cmake

```diff
@@ -0,0 +1,189 @@
+# Functions and macros to generate upgrade and downgrade scripts
+
+# concatenate_files(<output> <file> ...)
+#
+# Concatenate a list of files into <output>
+function(concatenate_files OUTPUT_FILE FIRST_FILE)
+  file(READ ${FIRST_FILE} _contents)
+  file(WRITE ${OUTPUT_FILE} "${_contents}")
+  foreach(_file ${ARGN})
+    file(READ ${_file} _contents)
+    file(APPEND ${OUTPUT_FILE} "${_contents}")
+  endforeach()
+endfunction()
```

Comment:
there is also a cat_files function in sql/CMakeLists.txt can we merge them/get rid of one of them?

Reply:
Agree, but I wanted to do that in a separate PR.